crates/python/src/qvm/api.rs

```diff
@@ -157,6 +178,19 @@ impl PyMultishotMeasureRequest {
             rng_seed,
         )))
     }
+
+    #[getter]
+    pub fn trials(&self) -> u16 {
+        self.as_inner().trials.get()
+    }
+
+    #[setter]
+    pub fn set_trials(&mut self, trials: u16) -> PyResult<()> {
+        // `NonZeroU16` doesn't implement `PyClass`, so `pyo3` doesn't allow it to be used
+        // as a method argument, even when combined with a `from_py_with` attribute.
```

Comment:
Out of curiosity, do you feel using `NonZeroU16` has made things easier or harder in this PR?

Reply:
Ha, I've been asking myself the same question. I mean, for this PR it definitely made things harder. Looking back, we should've created an issue for it and pressed on as I was just building on the existing API and this isn't exactly what we needed at the moment. I just thought it'd be simpler when I first got into it and didn't want to throw away the work once I got deep into it.

As far as using `NonZeroU16`, I'm torn. On one hand, it feels "correct" to be able to represent bounds as types. For this use case though, `NonZeroU16` added a lot of boiler plate and I'm not sure we gained a ton from it. We don't need to handle the error anymore, but now users have to import `NonZeroU16`, construct it, and handle the option just to use the function when they probably weren't going to consider passing `0` in to begin with, and the consequences of them doing so were handled with 3 lines of code. The pyo3 gymnastics needed was annoying, but I'm generally fine with that tradeoff if we're doing the right thing in the Rust crate. It's an interesting thing to reflect on though, because it makes you think about where the line should be drawn between types and implementation.